packages/components/src/components/stepper-horizontal/stepper-horizontal/stepper-horizontal.tsx

```diff
@@ -57,25 +56,11 @@ export class StepperHorizontal {
   private currentStepIndex: number;
 
   public connectedCallback(): void {
+    this.validateComponent();
```

Comment:
This is most likely to early in framework scenarios, especially Angular where properties are not bound, yet on first render. So this only reliably covers the reconnect scenario.
For the first render it would necessary to do this in `componentWillLoad` as it was before.
```suggestion
    this.validateComponent(); // on every reconnect
```